File: client/unity/Assets/Scripts/Assembly-CSharp/Chicken.cs
public class Chicken : AnimalDan
{
	public static int numChicken;

	public static AvPosition posNest;

	public Chicken(int id, sbyte species, sbyte cap)
		: base(0, 0, id, species)
	{
		captainID = cap;
		indexFood = 0;
		numChicken++;
	}

	public override void setInit()
	{
		posNext = new AvPosition();
		if (captainID == IDDB)
		{
			x = (xCur = (posNext.x = (FarmScr.numTileBarn + 3) * 24 + setX()));
			y = (yCur = (posNext.y = 72 + CRes.rnd(24) * 4));
			return;
		}
		updatePos();
		if (!LoadMap.isTrans(x, y))
		{
			setPosNext(new AvPosition((FarmScr.numTileBarn + 3) * 24 + setX(), 72 + CRes.rnd(24) * 4));
		}
		x = (xCur = posNext.x);
		y = (yCur = posNext.y);
	}

	public int setX()
	{
		int num = LoadMap.wMap - FarmScr.numTilePond - FarmScr.numTileBarn - 5;
		return CRes.rnd(num * 6) * 4;
	}

	public override void setFollowPos(AvPosition pos)
	{
		setPosNext(new AvPosition(pos.x - 48 + setX(), pos.y - 48 + CRes.rnd(24) * 4));
	}

	public override void setPos()
	{
		base.setPos();
	}
}

File: client/java/src/avt/Chicken.java
package avt;

public final class Chicken extends AnimalDan {
   public static int numChicken = 0;
   public static AvPosition s;

   public Chicken(int var1, byte var2, byte var3) {
      super(0, 0, var1, var2);
      super.captainID = 0;
      super.indexFood = 0;
      ++numChicken;
   }

   public final void setInit() {
      super.posNext = new AvPosition();
      if (super.captainID == super.IDDB) {
         super.x = super.xCur = super.posNext.x = (FarmScr.numTileBarn + 3) * 24 + randomX();
         super.y = super.yCur = super.posNext.y = 72 + (CRes.rnd(24) << 2);
      } else {
         this.updatePos();
         if (!LoadMap.isTrans(super.x, super.y)) {
            AvPosition var2 = new AvPosition((FarmScr.numTileBarn + 3) * 24 + randomX(), 72 + (CRes.rnd(24) << 2));
            super.posNext = var2;
         }

         super.x = super.xCur = super.posNext.x;
         super.y = super.yCur = super.posNext.y;
      }

   }

   private static int randomX() {
      return CRes.rnd((LoadMap.wMap - FarmScr.numTilePond - FarmScr.numTileBarn - 5) * 6) << 2;
   }

   public final void setFollowPos(AvPosition var1) {
      AvPosition var2 = new AvPosition(var1.x - 48 + randomX(), var1.y - 48 + (CRes.rnd(24) << 2));
      super.posNext = var2;
   }

   public final void setPos() {
      super.setPos();
   }
}
